backend/test/akvo/lumen/visualisation/map_test.clj

```diff
@@ -0,0 +1,177 @@
+          drop-table-seq
+          (format "DROP SEQUENCE IF EXISTS %s_rnum_seq CASCADE" table-name)
+          drop-table (jdbc/drop-table-ddl table-name)
+          drop-imported-table-seq
+          (format "DROP SEQUENCE IF EXISTS %s_rnum_seq CASCADE" imported-table-name)
+          drop-imported-table (jdbc/drop-table-ddl imported-table-name)]
+      (jdbc/db-do-commands *tenant-conn* [drop-table-seq
+                                          drop-table
+                                          drop-imported-table-seq
+                                          drop-imported-table])
+      (dataset/delete *tenant-conn* *dataset-id*))))
+
+
+(use-fixtures :once
+  system-fixture data-groups-future-fixture tenant-conn-fixture
+  error-tracker-fixture tu/spec-instrument dataset-data-fixture
+  dataset-id-fixture)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Utility fns, random data but we're using same form on the data and are
+;;; utilizing the form to get things like pointColourColumn and popupColumn
+;;;
+
+(defn popup
+  "Get one col from each group"
+  [columns-v4]
+  [{:column (->> columns-v4
+                 (filter (fn [{:keys [groupId key]}]
+                           (and (= groupId "metadata") (= key true))))
+                 first :id name)}
+   {:column (->> columns-v4
+                 (filter (fn [{:keys [groupId type]}]
+                           (and (= groupId "group1") (= type "number"))))
+                 first :id)}
+   {:column (->> columns-v4
+                 (filter (fn [{:keys [groupId type]}]
+                           (and (= groupId "group2") (= type "date"))))
+                 first :id)}])
+
+(defn geom
+  "Get the geopoint col"
+  [columns-v4]
+  (->> columns-v4 (filter #(= (:type %) "geopoint")) first :id))
+
+
+(defn pointColorColumn
+  "Get the column used as point colour column"
+  [columns-v4]
+  (->> columns-v4 (filter #(and (= (:groupId %) "group2")
+                                (= (:type %) "number")))
+       first :id))
+
+(defn layers
+  [dataset-id {:keys [columns-v4]}]
+  [{:aggregationMethod "avg"
+    :popup (popup columns-v4)
+    :filters []
+    :layerType "geo-location"
+    :legend {:title "Name", :visible true, :order {}}
+    :rasterId nil
+    :pointSize 3
+    :pointColorMapping []
+    :longitude nil
+    :datasetId dataset-id
+    :title "Random map"
+    :geom (geom columns-v4)
+    :pointColorColumn (pointColorColumn columns-v4)
+    :latitude nil
+    :visible true}])
+
+(defmulti verify-val (fn [_ f]
+                       (cond
+                         (inst? f) :inst
+                         (map? f) :geopoint ;; a bit simplistic :-)
+                         :else :default)))
+
+(defmethod verify-val :inst [v f]
+  ;; Since we convert datetime with timezone to text in the mapconfig sql
+  (let [custom-formatter (f/formatter "yyyy-MM-dd HH:mm:ssZ")]
+    (is (= (inst-ms (.toInstant (f/parse custom-formatter v)))
```

Comment:
I think it should be possible to use the JDK api [Instant](https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html) for this checks instead of using an external library.

Reply:
value `'2018-10-01 00:00:00+00'` received doesn't compliant with Instant/parse expectations so we're keeping current parse approach here

PS: value should has `T` instead of blank space